src/hooks/xcm/parachainApi/MoonbeamApi.ts

```diff
@@ -0,0 +1,201 @@
+import { LOCAL_STORAGE } from './../../../config/localStorage';
+import { EthereumProvider } from './../../types/CustomSignature';
+import { u8aToHex } from '@polkadot/util';
+import { decodeAddress } from '@polkadot/util-crypto';
+import BN from 'bn.js';
+import { supportEvmWallets, SupportWallet } from 'src/config/wallets';
+import { getTokenBal, setupNetwork } from 'src/config/web3';
+import moonbeamXcmAbi from 'src/config/web3/abi/moonbeam-xcm-abi.json';
+import { wait } from 'src/hooks/helper/common';
+import { Asset } from 'src/v2/models';
+import Web3 from 'web3';
+import { TransactionConfig } from 'web3-eth';
+import { AbiItem } from 'web3-utils';
+import { RelaychainApi } from '../SubstrateApi';
+import { isValidEvmAddress } from './../../../config/web3/utils/convert';
+import { getEvmProvider } from './../../helper/wallet';
+import { ethers } from 'ethers';
+
+type chainName = 'Moonriver' | 'Moonbeam';
+
+// Ref: https://docs.moonbeam.network/builders/build/canonical-contracts/precompiles/erc20/
+const NATIVE_TOKEN_ADDRESS = '0x0000000000000000000000000000000000000802';
+const PRE_COMPILED_ADDRESS = '0x0000000000000000000000000000000000000804';
+
+export class MoonbeamApi extends RelaychainApi {
+  private _AstarTokenId: { SDN: string; ASTR: string };
+  private _RpcEndpoint: { Moonriver: string; Moonbeam: string };
+  private _EvmId: { Moonriver: number; Moonbeam: number };
+  private _networkName: chainName;
+  private _web3: Web3 | null;
+  constructor(endpoint: string) {
+    super(endpoint);
+    this._networkName = endpoint.includes('moonriver') ? 'Moonriver' : 'Moonbeam';
+    // Todo: check the token address for ASTR
+    this._AstarTokenId = { SDN: '0xffffffff0ca324c842330521525e7de111f38972', ASTR: '0xToDo' };
```

Comment:
maybe all constant value should be defined in separate file?

Reply:
I've defined separately in the different block. https://github.com/AstarNetwork/astar-apps/pull/471/commits/d0d453d261e00a272668fae7a24d34663d16ad41